### ISSUE TITLE
Upload blobs to Azure storage in parallel

### ISSUE THREAD

**Issue by wwwjn** [codalab/codalab-worksheets#4286]
Upload blobs to azure storage has 2 steps:
1. write/put blob
2. commit blob

We can do first step (write/put) in parallel to reduce the total upload time.

**Comment by wwwjn** [comment]
Test result:
**TLDR: After using parallel, we can reduce the waiting time when some block is being committed.** 

In PR #4275, we minimize the size of MIN_WRITE_SIZE and MAX_WRITE_SIZE. This means when uploading a file, we need more put() operations and more commit() operations, which slow down the uploading process. So we need to parallel uploading process to speed up.

Measure the actual upload time of a file content:
1. without parallel: 17625s

![Image](https://user-images.githubusercontent.com/40016222/198908468-349c86c2-27e5-4036-9682-f6e8e384f2a7.png)



2. with parallel:  2690s

![Image](https://user-images.githubusercontent.com/40016222/198908457-99d799fd-e947-4d85-a6fc-11490e59d920.png)
